src/frontend/e2e/tests/upload.spec.ts

```diff
@@ -1,95 +1,105 @@
 import { test, expect } from "@playwright/test";
 import { uploadSampleFiles } from "../utils/upload";
 import { createSampleUploadData } from "../utils/sample";
-import { acceptSiteCookieTerms } from "../utils/common";
+import { BasePage } from "../pages/basePage";
+import { getLocations } from "../utils/common";
 
-const locations = [
-  "Africa/Angola/Luanda",
-  "Europe/Russia/Kaluga",
-  "Asia/China",
-];
+const locations = getLocations();
+const totalLocations = locations.length;
+
+let basePage: BasePage;
 test.describe("Upload sample tests", () => {
   const dateErrorMessage = "Update format to YYYY-MM-DD";
   const fileExtensions = ["fa", "fasta", "txt"]; //todo: add zip and gzip
   test.beforeEach(async ({ page }, workerInfo) => {
     const baseUrl = workerInfo.config.projects[0].use.baseURL;
     const url = `${baseUrl}/data/samples`;
-    await page.goto(url);
-    //accept site cookies
-    await acceptSiteCookieTerms(page);
+    basePage = new BasePage(page);
+    await Promise.all([basePage.gotoUrl(url), page.waitForNavigation()]);
+    page.waitForNavigation(),
```

Comment:
why do you need `waitForNavigation` here if you just waited for it in the previous line?